framework/Models.py

```diff
@@ -492,75 +509,51 @@ def _inputToInternal(self,dataIN,full=False):
            a mandatory key is the sampledVars'that contains a dictionary {'name variable':value}
-      @ Out, ([(inputDict)],copy.deepcopy(Kwargs)), tuple, return the new input in a tuple form
+      @ Out, ([(inputDict)],copy.deepcopy(kwargs)), tuple, return the new input in a tuple form
     """
     if len(myInput)>1:
       self.raiseAnError(IOError,'Only one input is accepted by the model type '+self.type+' with name'+self.name)
 
     inputDict   = self._inputToInternal(myInput[0])
     self._replaceVariablesNamesWithAliasSystem(inputDict,'input',False)
 
-    if 'SampledVars' in Kwargs.keys():
-      sampledVars = self._replaceVariablesNamesWithAliasSystem(Kwargs['SampledVars'],'input',False)
+    if 'SampledVars' in kwargs.keys():
+      sampledVars = self._replaceVariablesNamesWithAliasSystem(kwargs['SampledVars'],'input',False)
 
-    for key in Kwargs['SampledVars'].keys():
-      inputDict[key] = np.atleast_1d(Kwargs['SampledVars'][key])
+    for key in kwargs['SampledVars'].keys():
+      inputDict[key] = np.atleast_1d(kwargs['SampledVars'][key])
     for val in inputDict.values():
       if val is None:
         self.raiseAnError(IOError,'While preparing the input for the model '+self.type+' with name '+self.name+' found a None input variable '+ str(inputDict.items()))
     #the inputs/outputs should not be store locally since they might be used as a part of a list of input for the parallel runs
     #same reason why it should not be used the value of the counter inside the class but the one returned from outside as a part of the input
-    if 'SampledVars' in Kwargs.keys() and len(self.alias['input'].keys()) != 0:
-      Kwargs['SampledVars'] = sampledVars
-    return [(inputDict)],copy.deepcopy(Kwargs)
-
-  def updateInputFromOutside(self, Input, externalDict):
-    """
-      Method to update an input from outside
-      @ In, Input, list, list of inputs that needs to be updated
-      @ In, externalDict, dict, dictionary of new values that need to be added or updated
-      @ Out, inputOut, list, updated list of inputs
-    """
-    inputOut = Input
-    for key, value in externalDict.items():
-      inputOut[0][0][key] =  externalDict[key]
-      inputOut[1]["SampledVars"  ][key] =  externalDict[key]
-      inputOut[1]["SampledVarsPb"][key] =  1.0    #FIXME it is a mistake (Andrea). The SampledVarsPb for this variable should be transfred from outside
-      self._replaceVariablesNamesWithAliasSystem(inputOut[1]["SampledVars"  ],'input',False)
-      self._replaceVariablesNamesWithAliasSystem(inputOut[1]["SampledVarsPb"],'input',False)
-    return inputOut
-
-  def run(self,Input,jobHandler):
-    """
-      This method executes the model .
-      @ In,  Input, object, object contained the data to process. (inputToInternal output)
-      @ In,  jobHandler, JobHandler instance, the global job handler instance
-      @ Out, None
+    if 'SampledVars' in kwargs.keys() and len(self.alias['input'].keys()) != 0:
```

Comment:
Sounds good. I will add this change to my current local revision with this explanation. Good catch 👍